src/bin/cargo/commands/remove.rs

```diff
@@ -154,28 +254,72 @@ fn gc_workspace(workspace: &Workspace<'_>, dependencies: &[String]) -> CargoResu
     Ok(())
```

Comment:
Non-blocker: Just now noticed the unconditional write.  Should we track whether a change was made and only write if one happened?

Reply:
Sure. I added a `is_modified` flag which keeps track of this.

Note that there's sort of an oddity here now where instead of *always* removing empty `profile`/`patch`/`replace`/`workspace.dependencies` tables, it will only do so when another change is made elsewhere. We could try to move the `set_implicit` calls down to the level of the edit, but that would further complicate the code due to borrowing rules. If this seems like too weird of an issue, we can just revert and worry about that later/never.

Reply:
For me, removing tables without a cause would be a bug.